python/lsst/dax/distribution/DataIngest.py

```diff
@@ -0,0 +1,297 @@
+#!/usr/bin/env python3
+
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+
+import json
+import requests
+import subprocess
+
+
+class DataIngest():
+    """This class is used to communicate with the ingest system using
+    json formated requests and responses.
+
+    Parameters
+    ----------
+    host : str
+        Data ingest server host.
+    port : int
+        Data ingerst server port number.
+    user, auth_key : str, optional
+        User name and authorization key.
+    """
+
+    def __init__(self, host, port, user='', auth_key=''):
+        self._host = host
+        self._port = port
+        self._user = user
+        self._auth_key = auth_key
+        self._base_url = 'http://' + self._host + ':' + str(port) + '/'
+        print("base_url=", self._base_url)
+
+    def __repr__(self):
+        out = ("host=" + self._host + ":" + str(self._port)
+             + " user=" + self._user + "auth_key:****")
+        return out
+
+    def isIngestAlive(self):
+        """Test if ingest system is alive.
+        """
+        url = self._base_url + 'meta/version'
+        print("ingest url=", url)
+        response = requests.get(url)
```

Comment:
In all requests to REST services you should first check for the HTTP status code before drilling down onto the `JSON` response. Here is how this is done:
```
if response.status_code != 200: print "error: request failed with HTTP code: " + response.status_code 
```
Note, this applies to *ALL* requests, not just this one. Otherwise, you will get an exception at the next line in case of a error.